doi-transformer/src/main/java/no/unit/nva/doi/transformer/DataciteResponseConverter.java

```diff
@@ -0,0 +1,296 @@
+
+            publicationContext = new Journal.Builder()
+                .withPrintIssn(extractPrintIssn(dataciteResponse))
+                .withTitle(dataciteResponse.getContainer().getTitle())
+                .withOnlineIssn(extractOnlineIssn(dataciteResponse))
+                .withPeerReviewed(true)
+                .withOpenAccess(extractOpenAccess(dataciteResponse))
+                .withLevel(Level.NO_LEVEL)
+                .build();
+        }
+        return publicationContext;
+    }
+
+    private String extractOnlineIssn(DataciteResponse dataciteResponse) {
+        DataciteRelatedIdentifier result = extractIsPartOfRelations(dataciteResponse)
+            .stream()
+            .filter(this::isOnlineIssn)
+            .findAny()
+            .orElse(null);
+        return nonNull(result) ? IssnCleaner.clean(result.getRelatedIdentifier()) : null;
+    }
+
+    private boolean isOnlineIssn(DataciteRelatedIdentifier identifier) {
+        return DataciteRelatedIdentifierType.getByCode(identifier.getRelatedIdentifierType())
+            .equals(DataciteRelatedIdentifierType.EISSN);
+    }
+
+    private String extractPrintIssn(DataciteResponse dataciteResponse) {
+        DataciteRelatedIdentifier result = extractIsPartOfRelations(dataciteResponse)
+            .stream()
+            .filter(this::isPrintIssn)
+            .findAny()
+            .orElse(null);
+        return nonNull(result) ? IssnCleaner.clean(result.getRelatedIdentifier()) : null;
+    }
+
+    private List<DataciteRelatedIdentifier> extractIsPartOfRelations(DataciteResponse dataciteResponse) {
+        return dataciteResponse.getRelatedIdentifiers()
+            .stream()
+            .filter(this::isPartOf)
+            .collect(Collectors.toList());
+    }
+
+    private boolean isPrintIssn(DataciteRelatedIdentifier identifier) {
+        return DataciteRelatedIdentifierType.getByCode(identifier.getRelatedIdentifierType())
+            .equals(DataciteRelatedIdentifierType.ISSN);
+    }
+
+    private boolean isPartOf(DataciteRelatedIdentifier identifier) {
+        return DataciteRelationType.getByRelation(identifier.getRelationType())
+            .equals(DataciteRelationType.IS_PART_OF);
+    }
+
+    private boolean extractOpenAccess(DataciteResponse dataciteResponse) {
+        return Optional.ofNullable(dataciteResponse)
+            .map(DataciteResponse::getRightsList)
+            .stream()
+            .flatMap(Collection::stream)
+            .anyMatch(this::hasOpenAccessRights);
+    }
+
+    protected boolean hasOpenAccessRights(DataciteRights dataciteRights) {
+        return Optional.ofNullable(dataciteRights.getRightsUri())
+            .map(LicensingIndicator::isOpen).orElse(false);
+    }
+
+    private PublicationType extractPublicationType(DataciteResponse dataciteResponse) {
+        return DataciteTypesUtil.mapToType(dataciteResponse);
+    }
+
+    private String extractAbstract() {
```

Comment:
Set up a task to do something with the methods that return null in DataciteConverter